src/operator/tensor/cast_storage-inl.cuh

```diff
@@ -188,6 +188,15 @@ struct FillRspValsKernel {
   }
 };
 
+template<typename xpu, int ndim, typename DType>
+inline mshadow::Tensor<xpu, ndim, DType> AllocateTempDataForCast(const OpContext& op_ctx,
```

Comment:
https://books.google.co.uk/books?id=yQU-NlmQb_UC&lpg=PA72&vq=inline&dq=c%2B%2B+templates+the+complete+guide&pg=PA72&hl=en#v=onepage&q&f=false